proto/lbm/token/v1/query.proto

```diff
@@ -11,125 +11,237 @@ option go_package = "github.com/line/lbm-sdk/x/token";
   rpc Supply(QuerySupplyRequest) returns (QuerySupplyResponse) {
-    option (google.api.http).get = "/lbm/token/v1/supply/{class_id}";
+    option (google.api.http).get = "/lbm/token/v1/token_classes/{contract_id}/supply";
   }
 
-  // Token queries an token metadata based on its class id.
-  rpc Token(QueryTokenRequest) returns (QueryTokenResponse) {
-    option (google.api.http).get = "/lbm/token/v1/tokens/{class_id}";
+  // Minted queries the number of minted tokens from the given contract id.
+  rpc Minted(QueryMintedRequest) returns (QueryMintedResponse) {
+    option (google.api.http).get = "/lbm/token/v1/token_classes/{contract_id}/minted";
   }
 
-  // Tokens queries all token metadata.
-  rpc Tokens(QueryTokensRequest) returns (QueryTokensResponse) {
-    option (google.api.http).get = "/lbm/token/v1/tokens";
+  // Burnt queries the number of burnt tokens from the given contract id.
+  rpc Burnt(QueryBurntRequest) returns (QueryBurntResponse) {
+    option (google.api.http).get = "/lbm/token/v1/token_classes/{contract_id}/burnt";
   }
 
-  // Grants queries grants on a given grantee.
-  rpc Grants(QueryGrantsRequest) returns (QueryGrantsResponse) {
-    option (google.api.http).get = "/lbm/token/v1/grants/{grantee}/{class_id}";
+  // TokenClass queries an token metadata based on its contract id.
+  // if no token class found for the request, it wound return an error.
+  rpc TokenClass(QueryTokenClassRequest) returns (QueryTokenClassResponse) {
+    option (google.api.http).get = "/lbm/token/v1/token_classes/{contract_id}";
   }
 
-  // Approve queries approve on a given proxy approver pair.
-  rpc Approve(QueryApproveRequest) returns (QueryApproveResponse) {
-    option (google.api.http).get = "/lbm/token/v1/approve/{class_id}/{proxy}/{approver}";
+  // TokenClasses queries all token metadata.
+  // Since: finschia
+  rpc TokenClasses(QueryTokenClassesRequest) returns (QueryTokenClassesResponse) {
+    option (google.api.http).get = "/lbm/token/v1/token_classes";
   }
 
-  // Approves queries all approves on a given proxy.
-  rpc Approves(QueryApprovesRequest) returns (QueryApprovesResponse) {
-    option (google.api.http).get = "/lbm/token/v1/approves/{class_id}/{proxy}";
+  // Grant queries a permission on a given grantee permission pair.
+  // if no grant found for the request, it would return an error.
+  // Since: finschia
+  rpc Grant(QueryGrantRequest) returns (QueryGrantResponse) {
+    option (google.api.http).get = "/lbm/token/v1/token_classes/{contract_id}/grants/{grantee}/{permission}";
+  }
+
+  // GranteeGrants queries permissions on a given grantee.
+  // Since: finschia
+  rpc GranteeGrants(QueryGranteeGrantsRequest) returns (QueryGranteeGrantsResponse) {
+    option (google.api.http).get = "/lbm/token/v1/token_classes/{contract_id}/grants/{grantee}";
+  }
+
+  // Authorization queries authorization on a given operator holder pair.
+  // if no authorization found for the request, it would return an error.
```

Comment:
Could you write the comment to explain what kind of error clients will receive when no authorization is found?
Client need to distinguish such a not-found error from other kind of errors.

Reply:
I updated the comment.